File: 217.存在重复元素.cs
/*
 * @lc app=leetcode.cn id=217 lang=csharp
 *
 * [217] 存在重复元素
 */

// @lc code=start
public partial class Solution
{
    public bool ContainsDuplicate(int[] nums)
    {
        HashSet<int> visited = new HashSet<int>();
        for (int i = 0; i < nums.Length; i++)
        {
            if (visited.Contains(nums[i]))
            {
                return true;
            }
            visited.Add(nums[i]);
        }
        return false;
    }
}
// @lc code=end

File: src/217.存在重复元素.rs
/*
 * @lc app=leetcode.cn id=217 lang=rust
 *
 * [217] 存在重复元素
 */

// @lc code=start

use std::collections::HashSet;
impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut set = HashSet::new();
        for index in 0..nums.len() {
            if set.contains(&nums[index]) {
                return true;
            }
            set.insert(nums[index]);
        }
        return false;
    }
}
// @lc code=end